src/main/java/omero/gateway/facility/TransferFacilityHelper.java

```diff
@@ -125,84 +127,82 @@ List<File> downloadImage(SecurityContext context, String targetPath,
             throw new DSAccessException("Cannot retrieve original file", e);
         }
 
+        Map<Boolean, Object> result = new HashMap<Boolean, Object>();
```

Comment:
Do you need that Map? In the end only the `downloaded` list is returned. This map and the `notDownloaded` list isn't actuall used.

Reply:
I initially thought returning `downloaded `and `notDownloaded` (i.e. the result `Map`) but this would change the method definition (returns now a Map and not a list anymore) and may not be compatible with previous versions. 

The reason I kept them is beacuse I'm wondering if there is any possibility to return also the information contained in `notDownloaded`. Do you think it is possible ?

Reply:
I see. An option would be to add a second method for now which returns the map and mark the old one as deprecated. That would only be needed in the main TransferFacility class, not really in the helper. Looking at the code, I actually can't see a reason any more why I split the code into a helper class in the first place, could as well go into the TransferFacility class itself. Your thoughts @jburel  ? (Btw, this part of the Java gateway is not used by Insight, but should be!)

Reply:
But that could be done in a follow up PR anyway. Lets get this one in first :-)

Reply:
>  An option would be to add a second method for now which returns the map

Yes, exactly. It would then be possible to know which files had not been downloaded.

> But that could be done in a follow up PR anyway. Lets get this one in first :-)

Also true